components/supervisor/pkg/ports/ports.go

```diff
@@ -193,13 +193,21 @@ func (pm *Manager) updateState(exposed []ExposedPort, served []ServedPort, confi
 	}
 
 	if served != nil {
-		newServed := make(map[uint32]ServedPort)
+		var servedKeys []uint32 // to preserve insertion order
+		servedMap := make(map[uint32]ServedPort)
 		for _, port := range served {
-			current, exists := newServed[port.Port]
+			current, exists := servedMap[port.Port]
+			if !exists {
+				servedKeys = append(servedKeys, port.Port)
+			}
 			if !exists || (!port.BoundToLocalhost && current.BoundToLocalhost) {
-				newServed[port.Port] = port
+				servedMap[port.Port] = port
 			}
 		}
+		var newServed []ServedPort
+		for _, key := range servedKeys {
+			newServed = append(newServed, servedMap[key])
+		}
 		if !reflect.DeepEqual(pm.served, newServed) {
```

Comment:
Unrelated to this PR, but: Does it makes sense to `updateProxies` when only the order changes? Or do we want our custom test func here that checks for presence only?

Reply:
I don't think that order can change right now.